### PR TITLE
Added seqkit/sort

### LINKED ISSUE #4575: new module: seqkit/sort
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #4575 

- [x] Added seqkit/sort along with nf-tests.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/seqkit/sort/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,59 @@
+nextflow_process {
```
Comment (SPPearce) [inline modules/nf-core/seqkit/sort/tests/main.nf.test:1]: Can we have a test for a zipped fastq file please

**Review by SPPearce — COMMENTED** [review]
Looks pretty good, just a few suggestions

File: modules/nf-core/seqkit/sort/main.nf [hunk 2]
```diff
@@ -0,0 +1,60 @@
+process SEQKIT_SORT {
+    tag "$meta.id"
+    label 'process_medium'
```
[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/seqkit/sort/main.nf:3]: Thank you for the feedback.

Good point. According to docs (https://bioinf.shenwei.me/seqkit/usage/#parallelization-of-cpu-intensive-jobs), 4 threads are sufficient for sort as the file IO is the bottleneck. I'll downgrade it to 'process_low'.

File: modules/nf-core/seqkit/sort/main.nf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqkit:2.6.1--h9ee0642_0':
+        'biocontainers/seqkit:2.6.1--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(fastx)
+
+    output:
+    tuple val(meta), path("*.seqkit-sort.*"), emit: fastx
+    path "versions.yml"                     , emit: versions
```
[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/seqkit/sort/main.nf:15]: Yes, they are aligned.

File: modules/nf-core/seqkit/sort/main.nf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def extension = "fastq"
+    if ("$fastx" ==~ /.+\.fasta|.+\.fasta.gz|.+\.fa|.+\.fa.gz|.+\.fas|.+\.fas.gz|.+\.fna|.+\.fna.gz|.+\.fsa|.+\.fsa.gz/ ) {
+        extension = "fasta"
+    }
+    extension = fastx.toString().endsWith('.gz') ? "${extension}.gz" : extension
+    def call_gzip = extension.endsWith('.gz') ? '| gzip -c' : ''
```
[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/seqkit/sort/main.nf:28]: Yes, good point. I'll add that.

File: modules/nf-core/seqkit/sort/tests/main.nf.test (hunk 1 above)
Comment (GallVp) [reply modules/nf-core/seqkit/sort/tests/main.nf.test:1]: Yes, I'll add more tests.

File: modules/nf-core/seqkit/sort/main.nf [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    """
+    seqkit \\
+        sort \\
+        --threads $task.cpus \\
+        $args \\
+        $fastx \\
+        $call_gzip \\
+        > ${prefix}.seqkit-sort.${extension}
```
[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/seqkit/sort/main.nf:36]: Yes, I have changed that.

**Comment by GallVp** [comment]
Hi @SPPearce 

Thanks again for the feedback. Can you kindly review the changes?

File: modules/nf-core/seqkit/sort/main.nf [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    stub:
+    prefix          = task.ext.prefix ?: "${meta.id}"
+    def extension   = "fastq"
+    if ("$fastx" ==~ /.+\.fasta|.+\.fasta.gz|.+\.fa|.+\.fa.gz|.+\.fas|.+\.fas.gz|.+\.fna|.+\.fna.gz|.+\.fsa|.+\.fsa.gz/ ) {
+        extension   = "fasta"
+    }
+    extension       = fastx.toString().endsWith('.gz') ? "${extension}.gz" : extension
+    if("${prefix}.${extension}" == "$fastx") {
+        error "The name of the input file can't be the same as for the output prefix in the " +
+        "module SEQKIT_SORT (currently `$prefix`). Please choose a different one."
```
Comment (SPPearce) [inline modules/nf-core/seqkit/sort/main.nf:59]: What is the wording we use for e.g. samtools sort?

**Review by SPPearce — APPROVED** [review]
Can you check the wording on the name input/output please. I think we typically say how to change it (mention prefix). I'll check myself when I'm at my computer otherwise.

File: modules/nf-core/seqkit/sort/main.nf (hunk 6 above)
Comment (GallVp) [reply modules/nf-core/seqkit/sort/main.nf]: Thank you! I have changed the wording to make it similar to samtools/sort

**Pull request merged**